src/Flicking.ts

```diff
@@ -572,98 +580,190 @@ class Flicking extends Component {
   /**
    * Synchronize info of panels instance with info given by external rendering.
    * @ko 외부 렌더링 방식에 의해 입력받은 패널의 정보와 현재 플리킹이 갖는 패널 정보를 동기화한다.
-   * @param diffInfo - Info object of how panel elements are changed.<ko>패널의 DOM 요소들의 변경 정보를 담는 오브젝트.</ko>
-   * @param {HTMLElement[]} [diffInfo.list] - DOM elements list after update.<ko>업데이트 이후 DOM 요소들의 리스트</ko>
-   * @param {number[][]} [diffInfo.maintained] - Index tuple array of DOM elements maintained. Formatted with `[before, after]`.<ko>변경 전후에 유지된 DOM 요소들의 인덱스 튜플 배열. `[이전, 이후]`의 형식을 갖고 있어야 한다.</ko>
-   * @param {number[]} [diffInfo.added] - Index array of DOM elements added to `list`.<ko>`list`에서 추가된 DOM 요소들의 인덱스 배열.</ko>
-   * @param {number[]} [diffInfo.removed] - Index array of DOM elements removed from previous element list.<ko>이전 리스트에서 제거된 DOM 요소들의 인덱스 배열.</ko>
-   */
-  public sync(diffInfo: {
-    list: HTMLElement[],
-    maintained: number[][],
-    added: number[],
-    changed: number[][],
-    removed: number[],
-  }): this {
-    const { list, maintained, added, changed, removed } = diffInfo;
-
-    // Did not changed at all
-    if (added.length <= 0 && removed.length <= 0 && changed.length <= 0) {
-      return this;
-    }
-
+   * @param - Info object of how panel elements are changed.<ko>패널의 DOM 요소들의 변경 정보를 담는 오브젝트.</ko>
+   * @param - Info object of how the user's original rendering list is changed. This argument is only for the `renderOnlyVisible` option.<ko>사용자의 원본 렌더링 리스트의 변경 정보를 담는 오브젝트. 이 인자는 `renderOnlyVisible` 옵션을 위해서만 사용된다.</ko>
+   * @param - Clone count of the previous sync call. This argument is only for the `renderOnlyVisible` option.<ko>이전 sync() 호출 시점의 클론 횟수. 이 인자는 `renderOnlyVisible` 옵션을 위해서만 사용된다.</ko>
+   */
+  public sync(diffInfo: DiffResult<HTMLElement>, origListInfo?: DiffResult<any>, prevCloneCount: number = 0): this {
```

Comment:
@WoodNeck Implementation is too long to read code. It would be better to split function into meaningful unit.  But **Not Now**. It should be refactored later.

Reply:
@daybrush is working on it!